src/main/java/com/amazon/dlic/auth/http/jwt/HTTPExtensionJwtAuthenticator.java

```diff
@@ -0,0 +1,272 @@
+
+    private static final Pattern BASIC = Pattern.compile("^\\s*Basic\\s.*", Pattern.CASE_INSENSITIVE);
+    private static final String BEARER = "bearer ";
+
+    //TODO: TO SEE IF WE NEED THE FINAL FOR FOLLOWING
+    private JwtParser jwtParser;
+    private String subjectKey;
+
+    private String signingKey;
+    private String encryptionKey;
+
+    private Boolean bwcPluginCompatibilityMode;
+    public HTTPExtensionJwtAuthenticator() {
+        super();
+        init();
+    }
+
+    // FOR TESTING
+    public HTTPExtensionJwtAuthenticator(String signingKey, String encryptionKey,Boolean bwcPluginCompatibilityMode){
+        this.signingKey = signingKey;
+        this.encryptionKey = encryptionKey;
+        this.bwcPluginCompatibilityMode = bwcPluginCompatibilityMode;
+        init();
+    }
+
+    private void init() {
+
+        try {
+            if(signingKey == null || signingKey.length() == 0) {
+                log.error("signingKey must not be null or empty. JWT authentication will not work");
+            } else {
+
+                signingKey = signingKey.replace("-----BEGIN PUBLIC KEY-----\n", "");
+                signingKey = signingKey.replace("-----END PUBLIC KEY-----", "");
+
+                byte[] decoded = Decoders.BASE64.decode(signingKey);
+                Key key = null;
+
+                try {
+                    key = getPublicKey(decoded, "RSA");
+                } catch (Exception e) {
+                    log.debug("No public RSA key, try other algos ({})", e.toString());
+                }
+
+                try {
+                    key = getPublicKey(decoded, "EC");
+                } catch (Exception e) {
+                    log.debug("No public ECDSA key, try other algos ({})", e.toString());
+                }
+
+                if(key != null) {
+                    jwtParser = Jwts.parser().setSigningKey(key);
+                } else {
+                    jwtParser = Jwts.parser().setSigningKey(decoded);
+                }
+
+            }
+        } catch (Throwable e) {
+            log.error("Error creating JWT authenticator. JWT authentication will not work", e);
+            throw new RuntimeException(e);
+        }
+
+        subjectKey = "sub";
+    }
+
+    @Override
+    @SuppressWarnings("removal")
+    public AuthCredentials extractCredentials(RestRequest request, ThreadContext context) throws OpenSearchSecurityException {
+        final SecurityManager sm = System.getSecurityManager();
+
+        if (sm != null) {
+            sm.checkPermission(new SpecialPermission());
+        }
+
+        AuthCredentials creds = AccessController.doPrivileged(new PrivilegedAction<AuthCredentials>() {
+            @Override
+            public AuthCredentials run() {
+                return extractCredentials0(request);
+            }
+        });
+
+        return creds;
+    }
+
+    private AuthCredentials extractCredentials0(final RestRequest request) {
```

Comment:
Take a look at these classes from @scrawfor99 's PR, they do a good job isolating the specific functionality from the overall context.  While you might not need extra classes here, I think the parsing extracting logic would be cleaner

- [RestTokenExtractor](https://github.com/opensearch-project/OpenSearch/blob/e2269cbd40eeb44a9062991090387ca543639fef/server/src/main/java/org/opensearch/identity/tokens/RestTokenExtractor.java)
- [BasicAuthToken](https://github.com/opensearch-project/OpenSearch/blob/e2269cbd40eeb44a9062991090387ca543639fef/server/src/main/java/org/opensearch/identity/tokens/BasicAuthToken.java)

Reply:
@peternied and @DarshitChanpura , Thanks for the information and suggestions. Yes I think this is a good idea. We can do that as the follow-up PR for linking this backend with the configuration setup.